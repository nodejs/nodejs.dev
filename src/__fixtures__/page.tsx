import {
  PaginationInfo,
  LearnTemplateContext,
  NavigationData,
  PostTemplateData,
  PostTemplateContext,
  NodeReleaseData,
  PageTableOfContents,
  BlogCategory,
  BlogPost,
  ArticleData,
} from '../types';
import mockMDXBodyContent from './mockMDXBodyContent';

export const mockTableOfContents: PageTableOfContents = {
  items: [
    { title: 'mock-title', url: '#mock-title' },
    { title: 'mock-title-2', url: '#mock-title-2' },
  ],
};

export const createPaginationInfo = (): PaginationInfo =>
  ({
    slug: 'test-slug',
    title: 'test-title',
  } as PaginationInfo);

export const createNavigationSectionData = (): NavigationData =>
  ({
    'test-section': [
      {
        slug: 'test-slug-1',
        title: 'test-title-1',
      },
      {
        title: 'test-title-2',
        slug: 'test-slug-2',
      },
    ],
    'test-section2': [
      {
        slug: 'test-slug-3',
        title: 'test-title-3',
      },
      {
        title: 'test-title-4',
        slug: 'test-slug-4',
      },
    ],
  } as NavigationData);

export const createLearnPageData = () => ({
  data: {
    articleCurrentLanguage: {
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
      frontmatter: {
        title: 'test-title',
        description: 'test-description',
      },
      fields: {
        authors: ['test-user1', 'test-user2'],
      },
    } as ArticleData,
  },
});

export const createNodeReleasesData = (): NodeReleaseData[] =>
  [
    {
      endOfLife: '2022-04-30',
      ltsStart: '2019-10-21',
      maintenanceStart: '2020-11-30',
      initialRelease: '2019-04-23',
      codename: 'erbium',
      version: 'v12',
      fullVersion: 'v12.0.0',
      status: 'Maintenance LTS',
      isLts: false,
    },
    {
      endOfLife: '2023-04-30',
      ltsStart: '2020-10-17',
      maintenanceStart: '2021-10-19',
      initialRelease: '2020-04-21',
      codename: 'fermium',
      version: 'v14',
      fullVersion: 'v14.0.0',
      status: 'Active LTS',
      isLts: true,
    },
    {
      endOfLife: '2025-04-30',
      ltsStart: '2022-10-25',
      maintenanceStart: '2023-10-18',
      initialRelease: '2022-04-19',
      codename: 'v18',
      version: 'v18',
      fullVersion: 'v18.0.0',
      status: 'Current',
      isLts: false,
    },
  ] as NodeReleaseData[];

export const createLearnPageContext = (): LearnTemplateContext =>
  ({
    slug: 'test-slug',
    relativePath: 'test-path',
    next: createPaginationInfo(),
    previous: createPaginationInfo(),
    navigationData: createNavigationSectionData(),
  } as LearnTemplateContext);

export const createBlogPageContext = (): PostTemplateContext => ({
  slug: 'test-slug',
  relativePath: 'test-path',
  next: createPaginationInfo(),
  previous: createPaginationInfo(),
  navigationData: createNavigationSectionData(),
  recent: [
    {
      node: {
        frontmatter: {
          title: 'title-mock',
          category: {
            name: 'category-mock',
            slug: 'category-mock-slug',
          },
          blogAuthors: [
            {
              id: 'id-mock',
              name: 'name-mock',
              website: 'url-mock',
            },
          ],
        },
        fields: {
          date: '11/11/2022',
          slug: 'slug-mock',
          readingTime: { text: 'text-mock' },
        },
      },
    },
  ],
});

export const createBlogData = (): {
  posts: { edges: BlogPost[] };
  categories: { edges: BlogCategory[] };
} => ({
  posts: {
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Mock blog title',
            category: {
              name: 'mock-category',
              slug: 'Mock Category Slug',
            },
            blogAuthors: [
              {
                name: 'Mock author name',
                website: 'Mock URL',
              },
            ],
          },
          fields: {
            date: '11/11/2022',
            slug: 'Mock blog slug',
            readingTime: {
              text: '1 min read',
            },
          },
        },
      },
    ],
  },
  categories: {
    edges: [],
  },
});

export const createBlogPageData = (): PostTemplateData => ({
  mdx: {
    body: mockMDXBodyContent,
    excerpt: 'excerpt-mock',
    frontmatter: {
      title: 't,itle-mock',
      blogAuthors: [
        {
          id: 'id-mock',
          name: 'name-mock',
          website: 'url-mock',
        },
      ],
    },
    fields: { slug: 'slug-mock', date: '11/11/2022' },
  },
});

export const createGeneralPageData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['author-mock'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createResourcesData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['MrJithil'],
      },
      frontmatter: {
        description: 'Mock Description',
        title: 'Mock Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});

export const createPrivacyData = () => ({
  data: {
    articleCurrentLanguage: {
      fields: {
        authors: ['Author'],
      },
      frontmatter: {
        description: 'Description',
        title: 'Title',
      },
      body: mockMDXBodyContent,
      tableOfContents: mockTableOfContents,
    } as ArticleData,
  },
});
