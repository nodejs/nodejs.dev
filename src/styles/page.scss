.page > .article-reader {
  margin: 0 auto;
}
