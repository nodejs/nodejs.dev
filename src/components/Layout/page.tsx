import React from 'react';
import Article from '../Article';
import Layout from '.';

import '../../styles/page.scss';

interface Page {
  editPath?: string;
  html: string;
  tableOfContents: string;
  title: string;
  description: string;
  authors: string[];
  location?: Location;
}

export default function PageLayout(props: Page): JSX.Element {
  const {
    editPath,
    html,
    tableOfContents,
    title,
    description,
    authors,
    location,
  } = props;
  return (
    <Layout title={title} location={location} description={description}>
      <main className="page">
        <Article
          html={html}
          title={title}
          authors={authors}
          editPath={editPath}
          tableOfContents={tableOfContents}
        />
      </main>
    </Layout>
  );
}
