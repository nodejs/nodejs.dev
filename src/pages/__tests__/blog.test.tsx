import React from 'react';
import { render } from '@testing-library/react';
import AllBlogPosts from '../blog';
import { createBlogData } from '../../__fixtures__/page';

const mockData = createBlogData();

describe('Blog page', () => {
  it('renders correctly', () => {
    const { container } = render(<AllBlogPosts data={mockData} />);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('renders correctly for empty blogs list', () => {
    const { container } = render(
      <AllBlogPosts
        data={{
          posts: {
            edges: [],
          },
          categories: {
            edges: [],
          },
        }}
      />
    );
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });
});
