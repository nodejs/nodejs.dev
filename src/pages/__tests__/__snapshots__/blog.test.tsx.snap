// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog page renders correctly 1`] = `
<main
  class="grid-container"
>
  <nav
    class="sideNav"
  >
    <button
      class="sideNavOpen"
      type="button"
    >
      Menu
    </button>
    <div
      class="sideNavSection"
    >
      <div
        class="sideNavSectionTitle"
      >
        <b>
          Blog Categories
        </b>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-8rv7ug-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </nav>
  <div
    class="blogGridContainer"
  >
    <div
      class="blogCategoryHeader"
    >
      <h2>
        Node.js Blog
      </h2>
      <span>
        The latest Node.js news, case studies, tutorials, and resources.
      </span>
    </div>
    <div
      class="blogItems"
    >
      <div
        class="blogCard"
      >
        <div
          class="title"
        >
          <a
            href="Mock blog slug/"
          >
            Mock blog title
          </a>
          <div
            class="metadata"
          >
            <a
              class="category"
              href="/blog/mock-category/"
            >
              Mock Category Slug
            </a>
            <span>
              1 min read
            </span>
          </div>
        </div>
        <div
          class="content"
        >
          <h4>
            11/11/2022
          </h4>
          <p>
            by
             
            <span>
              Mock author name
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`Blog page renders correctly for empty blogs list 1`] = `
<main
  class="grid-container"
>
  <nav
    class="sideNav"
  >
    <button
      class="sideNavOpen"
      type="button"
    >
      Menu
    </button>
    <div
      class="sideNavSection"
    >
      <div
        class="sideNavSectionTitle"
      >
        <b>
          Blog Categories
        </b>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-8rv7ug-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </nav>
  <div
    class="blogGridContainer"
  >
    <div
      class="blogCategoryHeader"
    >
      <h2>
        Node.js Blog
      </h2>
      <span>
        The latest Node.js news, case studies, tutorials, and resources.
      </span>
    </div>
    <div
      class="blogItems"
    />
  </div>
</main>
`;
